helix-core/src/wrap.rs

```diff
@@ -0,0 +1,350 @@
+use ropey::RopeSlice;
+
+/// Given a slice of text, return the text re-wrapped to fit
+/// it within the given width.
+pub fn reflow_hard_wrap(text: RopeSlice, max_line_len: usize) -> String {
```

Comment:
Have you evaulated simply using https://github.com/mgeisler/textwrap which will also solve for cases like multiwidth characters?

Reply:
Not yet! That could clean up most of the logic. I'll definitely check it out.

Reply:
Looked at it a bit, and it seems like the right approach. Thanks for the suggestion. I wish I'd asked around a bit earlier, but I wanted to delete most of this code anyway.

I'm especially impressed with the line-breaking algorithms they have. Not sure if using the "optimal" line breaking algorithm is worth the extra binary size. Personally, I'd want it for the extra ~50KB. Thoughts?